File: range-sum-query-immutable/range-sum-query-immutable.cs
public class NumArray {
    private int[] sums;
    public NumArray(int[] nums) {
        sums = new int[nums.Length + 1];
        for(int i = 1; i < sums.Length; i++) {
            sums[i] = sums[i - 1] + nums[i - 1];
        }
    }
    
    public int SumRange(int left, int right) {
        return sums[right + 1] - sums[left];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(left,right);
 */

File: range-sum-query-immutable/range-sum-query-immutable.java
class NumArray {
    private int[] _prefixSum;
    public NumArray(int[] nums) {
        _prefixSum = new int[nums.length + 1];
        for(int i = 0; i < nums.length; i++) {
            _prefixSum[i + 1] = nums[i] + _prefixSum[i];
        }
    }
    
    public int sumRange(int left, int right) {
        return _prefixSum[right + 1] - _prefixSum[left];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */